script/reproducible/verify.bat

```diff
@@ -0,0 +1,129 @@
+@ECHO OFF
+set CHECK_VERSION=%1
+SHIFT
+
+set buildir=C:\projects\
+
+REM Interpret arguments
+:loop
+IF NOT "%1"=="" (
+    IF "%1"=="-buildir" (
+        SET buildir=%2
+        SHIFT
+    )
+    SHIFT
+    GOTO :loop
+)
+
+@ECHO ON
+
+set ELECTRON_OUT_DIR=Default
+set ELECTRON_ENABLE_STACK_DUMPING=1
+
+REM Check for disk space
+Rem        543210987654321
+Set "Blank=               "
+Set "GB100=   107374182400"
+
+for /f "tokens=2" %%A in (
+  'wmic LogicalDisk Get DeviceID^,FreeSpace ^| find /i "C:"'
+) Do Set "FreeSpace=%Blank%%%A"
+Set "FreeSpace=%FreeSpace:~-15%"
+
+Echo FreeSpace="%FreeSpace%"
+Echo    100 GB="%GB100%"
+
+If "%FreeSpace%" gtr "%GB100%" (
+  Echo yes enough free space
+) else (
+  Echo not enough free space - 100GB 
+  exit 5
+)
+
+REM Install missing dependencies
+powershell -command "& { iwr https://raw.githubusercontent.com/electron/electron/%CHECK_VERSION%/script/setup-win-for-dev.bat -OutFile C:\install.bat }"
+cmd /c C:\install.bat
+
+
+set GN_CONFIG=release
+
+echo "Building $env:GN_CONFIG build"
+SET PATH=%PATH%;C:\depot_tools\
+SET PATH=%PATH%;C:\Python27\
+git config --global core.longpaths true
+if not exist %buildir% mkdir %buildir%
+cd %buildir%
+set CHROMIUM_BUILDTOOLS_PATH=%buildir%\src\buildtools
+set SCCACHE_PATH=%buildir%\src\electron\external_binaries\sccache.exe

```

Comment:
I'm confused why sccache would be involved in a reproducibility test. Shouldn't we be building from scratch both times?

Reply:
So, since the cache seems to be a major offender in this particular case, maybe 1 build without cache, 2 with the cache should be better.

What we want to actually test is that build is:
1. local reproducible when built twice in the same setup of the remote machine (so sccache enabled)
2. that sccache is not introducing weird caching strategies that breaks the correctness of the executable.

Reply:
it's fine to use sccache in the build i suppose, as long as the cache is cleared between builds. Otherwise it's not really testing reproducibility, right? Since it's just reusing the build outputs.